module/zfs/vdev.c

```diff
@@ -1688,8 +1693,15 @@ vdev_open(vdev_t *vd)
 	 * since this would just restart the scrub we are already doing.
 	 */
 	if (vd->vdev_ops->vdev_op_leaf && !spa->spa_scrub_reopen &&
-	    vdev_resilver_needed(vd, NULL, NULL))
-		spa_async_request(spa, SPA_ASYNC_RESILVER);
+	    vdev_resilver_needed(vd, NULL, NULL)) {
+		if (dsl_scan_resilvering(spa->spa_dsl_pool) &&
+		    spa_feature_is_enabled(spa, SPA_FEATURE_RESILVER_DEFER)) {
+			vd->vdev_resilver_deferred = B_TRUE;
+			spa->spa_resilver_deferred = B_TRUE;
+		} else {
+			spa_async_request(spa, SPA_ASYNC_RESILVER);
```

Comment:
This chunk of code is repeated a few times.  Maybe make a helper function for it?  (which maybe we could also use in spa_vdev_attach())

Reply:
ping